### PR TITLE
Added conditions, and Input class.

### PR DESCRIPTION
**Pull request by Mifou**
And also addapted game to play with hot seat mode for 2 players

### CODE REVIEW

File: src/TicTacToe/EndingConditions.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    public EndingConditions (Board board){
+        this.board = board;
+    }
+
+    public boolean isDraw(){
+        char elements[][] = board.getElements();
+        for (int i = 0; i < elements.length; i++) {
+            for (int j = 0; j < elements[i].length; j++) {
+                if (elements[i][j] >= '1'&& elements[i][j] <='9' || isWin()) {
```
Comment (caroo7) [inline src/TicTacToe/EndingConditions.java]: Chock isWin condition before all loops to skip going inside loops in that case.

File: src/TicTacToe/EndingConditions.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                if ((elements[0][j] == elements[1][j]) && (elements[0][j] == elements[2][j])) {
+                    return true;
+                }
+            }
+
+        }
+        //In diagonal case
+        for (int i = 0; i < elements.length; i++) {
+            for (int j = 0; j < elements[i].length; j++) {
+                if ((elements[0][0] == elements[1][1]) && (elements[0][0] == elements[2][2])) {
```
Comment (caroo7) [inline src/TicTacToe/EndingConditions.java]: What happen if you have situation like that:
O X O
X O X
O X X
;)

File: src/TicTacToe/Input.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            }
+        } while (field < 1 || field > 9);
+
+        return field;
+
+    }
+
+    public boolean isSignIn(int position) {
+        char elements[][] = board.getElements();
+        position -= 1;
```
Comment (caroo7) [inline src/TicTacToe/Input.java]: Maybe add some comment here why you need to do this.

File: src/TicTacToe/Main.java [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
+            while (input.isSignIn(position)) {
+                System.out.println("There is a sign in this field, chose free field");
+                position = input.chosePosition();
+            }
+
 
+            actualPlayer.move(position);
 
+            board.displayBoard();
+            if (endingConditions.isWin()) {
```
Comment (caroo7) [inline src/TicTacToe/Main.java]: You can move it to separate method checkEndingConditions.

File: src/TicTacToe/Main.java [hunk 5]
```diff
@@ -1,19 +1,46 @@
...
+            }
+            if (endingConditions.isDraw()) {
+                break;
 
-        board.displayBoard();
-        Jack.move(5);
-        Bob.move(5);
-        board.displayBoard();
+            }
+            if (actualPlayer.equals(jack)) {
```
Comment (caroo7) [inline src/TicTacToe/Main.java]: You can move it to separate method changeActualPlayer

File: src/TicTacToe/EndingConditions.java (hunk 1 above)
Comment (Mifou) [reply src/TicTacToe/EndingConditions.java]: done

File: src/TicTacToe/EndingConditions.java (hunk 2 above)
Comment (Mifou) [reply src/TicTacToe/EndingConditions.java]: done

File: src/TicTacToe/Input.java (hunk 3 above)
Comment (Mifou) [reply src/TicTacToe/Input.java]: done

File: src/TicTacToe/Main.java (hunk 4 above)
Comment (Mifou) [reply src/TicTacToe/Main.java]: done

File: src/TicTacToe/Main.java (hunk 5 above)
Comment (Mifou) [reply src/TicTacToe/Main.java]: done

**Pull request merged**